ldmud: init at 3.6.6

cpu opened the pull request:
###### Description of changes

LDMud is a game engine for text based multi-user dungeon games ([MUDs](https://en.wikipedia.org/wiki/MUD)), in the ["LPMud" heritage](https://en.wikipedia.org/wiki/LPMud). In essence it is a virtual machine that allows developers to implement MUD games ("mudlibs" in LD parlance) using the LPC programming language.

Its homepage is: http://www.ldmud.eu/
Source code available at: https://github.com/ldmud/ldmud

In the project's own words:
> This is 'LDMud', a gamedriver for LPMuds. (LPC compiler, interpreter
> and runtime environment.)
>
> LDMud started as a project to clean up and modernize Amylaar's LPMud
> gamedriver. Primary goals are full documentation, a commented source
> body and out-of-the-box support for the major mudlibs, of which the
> commented source body has been pretty much completed. During the course
> of work a lot of bug fixes and improvements found their way into the
> driver - much more than originally expected, and definitely enough to
> make LDMud a driver in its own right.

######  Nixpkgs Derivation Notes 

For nixpkgs the new LDMud addition is largely a standard derivation for an autoreconf/automake based project. Some small tweaks are required to get everything lined up between the Nix environment and the LDMud build system.

The new LDMud derivation is placed in `pkgs/games/ldmud/` alongside other MUD-related projects in the games category (e.g. `mudlet`, `blightmud`).

The License is set to `licenses.unfreeRedistributable` since the [upstream COPYRIGHT](https://github.com/ldmud/ldmud/blob/master/COPYRIGHT) is non-standard but does allow redistribution. The LDMud copyright grants the rights to redistribute unmodified or modified binaries or source, with the added requirement that the software not be used for monetary gain. We include the COPYRIGHT file in the built output to ensure we meet the requirements for distribution.

LDMud offers a variety of optional features, some of which require heavier dependencies (e.g. MySQL/Postgres client libraries). To support both a minimal default build equal to what one gets using the upstream build with no customization, and to also support a more useful fully featured build, the LDMud derivation is added to `all-packages.nix` twice:

1. the `ldmud` attribute builds the minimal default configuration (just the optional gcrypt and pcre options enabled, matching upstream).
2. the `ldmud-full` attribute builds LDMud with all of the optional features enabled (database support, Python support, TLS support, etc).

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [X] (Package updates) Added a release notes entry if the change is major or breaking
  - [X] (Module updates) Added a release notes entry if the change is significant
  - [X] (Module addition) Added a release notes entry if adding a new NixOS module
  - [X] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

cpu committed:
ldmud: init at 3.6.6

LDMud is a game engine for text based multi-user dungeon games (MUDs),
in the "LP" heritage. In essence it is a virtual machine that allows
developers to implement MUD games ("mudlibs" in LD parlance) using the
LPC programming language.

In the project's own words:
> This is 'LDMud', a gamedriver for LPMuds. (LPC compiler, interpreter
> and runtime environment.)
>
> LDMud started as a project to clean up and modernize Amylaar's LPMud
> gamedriver. Primary goals are full documentation, a commented source
> body and out-of-the-box support for the major mudlibs, of which the
> commented source body has been pretty much completed. During the course
> of work a lot of bug fixes and improvements found their way into the
> driver - much more than originally expected, and definitely enough to
> make LDMud a driver in its own right.

For nixpkgs the new LDMud addition is largely a standard derivation for
an autoreconf/automake based project. Some small tweaks are required to
get everything lined up between the Nix environment and the LDMud build
system.

The new LDMud derivation is placed in `pkgs/games/ldmud/` alongside
other MUD-related projects in the games category (e.g. `mudlet`,
`blightmud`).

LD offers a variety of optional features, some of which require heavier
dependencies (e.g. MySQL/Postgres client libraries). To support both
a minimal default build equal to what one gets using the upstream build
with no customization, and to also support a more useful fully featured
build, the LDMud derivation is added to `all-packages.nix` twice:

1. the `ldmud` attribute builds the minimal default configuration (just
   the optional gcrypt and pcre options enabled, matching upstream).
2. the `ldmud-full` attribute builds LDMud with all of the optional
   features enabled (database support, Python support, TLS support,
   etc).

SuperSandro2000 commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,107 @@
+, postgresSupport ? false, postgresql, sqliteSupport ? false, sqlite
+, jsonSupport ? false, json_c, xmlSupport ? false, libxml2
+, deprecatedSupport ? true, tlsSupport ? false, openssl, pythonSupport ? false
+, python39, pythonSupportVersion ? python39 }:
```
```suggestion
, python39 }:
```
We can't use python 3.10?

cpu replied:
There was an upstream regression in 3.10 that caused me some trouble when I first started this derivation, but that was months ago and I reported the bug upstream and it was fixed. I will swap this out for python310 and do some testing to make sure it's working as intended.

In the spirit of your other feedback I'll drop the `pythonSupportVersion` and just use python310 implicitly.

SuperSandro2000 commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,107 @@
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-2TaFt+T9B5Df6KWRQcbhY1E1D6NISb0oqLgyX47f5lI=";
+    name = pname;
```
```suggestion
```

abathur replied:
It seems like the justification for this is below:

```
   # LD keeps its source code in a 'src' subfolder. Note the 'sourceRoot' is
   # relative to the tmp build directory nix creates so we need the predictable
   # 'name' set above.
   sourceRoot = "${pname}/src";
```

What's the more-correct pattern for getting the right sourceRoot? I see some packages using `sourceRoot = "source/src";`

cpu replied:
I mentioned this in the comment on `sourceRoot` below, it's necessary to set the `name` explicitly to `pname` in order for the `sourceRoot` to be correct. When I delete just this line I get an error during build:
```
error: builder for '/nix/store/glgi1v6fkg90ip1y556p0aa3bqy8k7f7-ldmud-3.6.6.drv' failed with exit code 1;
       last 4 log lines:
       > unpacking sources
       > unpacking source archive /nix/store/qn3lv2qnflkp0qsqsakwq2hzcsdmb8ds-source
       > source root is ldmud/src
       > chmod: cannot access 'ldmud/src': No such file or directory
```

Similarly not setting `sourceRoot` or using `sourceRoot = "src";` doesn't work in my testing.

cpu replied:
>  I see some packages using sourceRoot = "source/src";

Ah! That seems workable. Thanks @abathur I'll go with that vs setting `name`.

SuperSandro2000 commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,107 @@
+  # 'name' set above.
+  sourceRoot = "${pname}/src";
+
+  # Autoreconf, pkg-config and bison are required in all cases.
+  # Other dependencies are optional.
```
```suggestion
```

SuperSandro2000 commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,107 @@
+  # Autoreconf, pkg-config and bison are required in all cases.
+  # Other dependencies are optional.
+  nativeBuildInputs = [ autoreconfHook pkg-config bison ]
+    ++ lib.optional pcreSupport pcre ++ lib.optional gcryptSupport libgcrypt
+    ++ lib.optional mccpSupport zlib ++ lib.optional mysqlSupport libmysqlclient
```
```suggestion
    ++ lib.optional pcreSupport pcre
    ++ lib.optional gcryptSupport libgcrypt
    ++ lib.optional mccpSupport zlib
    ++ lib.optional mysqlSupport libmysqlclient
```

SuperSandro2000 commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,107 @@
+    # here and so default to openssl.
+    ++ lib.optional tlsSupport openssl
+    ++ lib.optional pythonSupport pythonSupportVersion;
+  # On Darwin we need to include libiconv explicitly or there will be build errors
+  # about missing symbols.
```
```suggestion

```

SuperSandro2000 commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,107 @@
+    ++ lib.optional pcreSupport pcre ++ lib.optional gcryptSupport libgcrypt
+    ++ lib.optional mccpSupport zlib ++ lib.optional mysqlSupport libmysqlclient
+    ++ lib.optional postgresSupport postgresql
+    ++ lib.optional sqliteSupport sqlite ++ lib.optional jsonSupport json_c
```
```suggestion
    ++ lib.optional sqliteSupport sqlite
    ++ lib.optional jsonSupport json_c
```

SuperSandro2000 commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,107 @@
+{ lib, fetchFromGitHub, stdenv, autoreconfHook, pkg-config, bison, libiconv
+, pcreSupport ? true, pcre, gcryptSupport ? true, libgcrypt, erqSupport ? "erq"
+, compatModeSupport ? false, filenameSpacesSupport ? false, ipv6Support ? false
+, mccpSupport ? false, zlib, mysqlSupport ? false, libmysqlclient
+, postgresSupport ? false, postgresql, sqliteSupport ? false, sqlite
+, jsonSupport ? false, json_c, xmlSupport ? false, libxml2
+, deprecatedSupport ? true, tlsSupport ? false, openssl, pythonSupport ? false
```
Do we really need all those options? No one is going to test them in all combinations. How about we make some good default assumptions and drop half of those options?

cpu replied:
That's fair feedback. I will rework and drop some low hanging fruit like `deprecatedSupport`, and `filenameSpacesSupport`.

cpu replied:
I've trimmed it down to just:
* MCCP support
* Mysql support
* Postgres support
* Sqlite support
* TLS support
* Python support
* IPv6 support

I dropped the configurable erq mode, the deprecated efuns option, the fielname spaces option, gcrypt option, pcre option, and compat mode option.

SuperSandro2000 commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,107 @@
+  # when configure.ac is in the expected place in the srcroot. Symlink to the
+  # rescue.
+  preAutoreconf = ''
+    ln --force --symbolic ./autoconf/configure.ac ./
```
```suggestion
    ln -fs ./autoconf/configure.ac ./
```

SuperSandro2000 commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,107 @@
+  # about missing symbols.
+  buildInputs = lib.optionals stdenv.isDarwin [ libiconv ];
+
+  # To support systems without autoconf LD puts its `configure.ac` in a subdir
+  # and uses a small wrapper script, `src/autogen.sh` to process it. This script
+  # checks if autoconf is installed and either runs it and copies the generated
+  # files to the src root, or it copies pre-generated vendored copies to the src
+  # root.
+  #
+  # We always have autoconf and a nice stdenv autoreconfHook we can use, but only
+  # when configure.ac is in the expected place in the srcroot. Symlink to the
+  # rescue.
```
Can we short this comment to one or two lines?

cpu replied:
Definitely.

SuperSandro2000 commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,107 @@
+    (lib.enableFeature pythonSupport "use-python")
+  ];
+
+  # We need to help LD find MySQL client headers/libs when mysqlSupport is
+  # enabled.
```
```suggestion

```

SuperSandro2000 commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,107 @@
+    export LDFLAGS="-L${libmysqlclient}/lib/mysql"
+  '';
+
+  installTargets = "install-driver install-utils install-headers";
```
```suggestion
  installTargets = [ "install-driver" "install-utils" "install-headers" ];
```

SuperSandro2000 commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,107 @@
+  installTargets = "install-driver install-utils install-headers";
+
+  meta = with lib; {
+    description =
+      "LDMud is a gamedriver for LPMuds. (LPC compiler, interpreter and runtime environment.)";
```
```suggestion
    description = "LDMud is a gamedriver for LPMuds. (LPC compiler, interpreter and runtime environment.)";
```

SuperSandro2000 commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,107 @@
+    description =
+      "LDMud is a gamedriver for LPMuds. (LPC compiler, interpreter and runtime environment.)";
+    homepage = "https://ldmud.eu";
+    changelog = "https://github.com/${pname}/${pname}/blob/${version}/HISTORY";
```
```suggestion
    changelog = "https://github.com/ldmud/ldmud/blob/${version}/HISTORY";
```

SuperSandro2000 commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,107 @@
+, deprecatedSupport ? true, tlsSupport ? false, openssl, pythonSupport ? false
+, python39, pythonSupportVersion ? python39 }:
+
+let supportedErq = [ "no" "erq" "xerq" ];
+in assert builtins.elem erqSupport supportedErq;
```
```suggestion
assert builtins.elem erqSupport [ "no" "erq" "xerq" ];
```

cpu replied:
This fell away because I removed the option to configure the erq mode and simply defaulted to `xerq`.

SuperSandro2000 reviewed (commented):
I suggested to drop some comments which are not required in my opinion since they mostly just describe the code below them

cpu commented:
@SuperSandro2000 I think I've addressed all of your feedback except for the one unresolved comment about setting `name` to `pname`. Thanks for the review!

cpu commented:
> I think I've addressed all of your feedback except for the one unresolved comment about setting name to pname. 

That's taken care of as well as of [283f997](https://github.com/NixOS/nixpkgs/pull/190682/commits/283f9974fa61c1622bcb77190e06e59c521b1e7c).

(_If you'd prefer I squash my own commits and force push let me know, operating under the assumption whoever merges will squash-merge it._)

viraptor commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,75 @@
+      a driver in its own right.
+    '';
+    # See https://github.com/ldmud/ldmud/blob/master/COPYRIGHT
+    license = licenses.free;
```
At the risk of starting "one of those debates". `free` is for "Unspecified free software license". I expect that's following the FSF/OSI/etc definition where "noncommercial" (https://github.com/ldmud/ldmud/blob/64b3588a13fb0c761da62d6deb56dfa380b03c6f/COPYRIGHT#L13-L14) licenses are not "free" (https://en.wikipedia.org/wiki/Free-software_license)

Should this be its own license instead?

cpu replied:
@viraptor This is definitely outside my wheelhouse and I could use some advice. I think there are three options:

* Keep it as is
  * _Upsides:_ the upstream license allows redistribution which seems most important for the nixpkgs context, and we don't litter the licenses metadata with a one-off license just for LD.
  * _Downsides:_ the upstream license isn't FSF blessed and the non-commercial aspect isn't well represented in the package metadata.
* Add a special license just for LDMud (`licenses.ldmud`?)
  * _Upsides:_ fully captures the LDMud specific constraints from the upstream copyright.
  * _Downsides:_ I can't image other packages will use this license, so there's some Nixpkgs-wide bloat.
* Switch it to a non-free package
  * _Upsides:_ resolves the situation without Nixpkgs bloat.
  * _Downsides:_ other than the non-commercial constraint LDMud is what I'd consider free software in the sense we can modify it, redistribute it, etc. It would be a shame to have no binary builds and an extra hurdle for users to jump through
  
What do others think? Is there someone in particular well suited to give guidance here we could tag in?

Thanks!

dali99 replied:
A license is just an attribute set with a pretty name. if you need a custom one [you can do something like what's done in n8n](https://github.com/NixOS/nixpkgs/blob/ac3e43ca8a4f65372657bd9b5c5a28b66921f189/pkgs/applications/networking/n8n/default.nix#L29-L35)

It's not commercially redistributable so it'd just be the same as unfree. (And it doesn't matter much since hydra doesn't build unfree things even if they are redistributable)

I'd just set it to `unfree`

abathur replied:
I don't have much context on the nixpkgs imperatives here, but at least from reading through https://github.com/NixOS/nixpkgs/issues/83884 it seems like it's probably worth the effort to mark it as redistributable, since it would enable the existing ~unofficial caches to build it for now and would ensure it starts getting built if/when the same applies to nixpkgs itself.

dali99 replied:
if it is redistributable one can set unfreeRedistributable. I'm not sure if it is however.

abathur replied:
The upstream's [COPYRIGHT](https://github.com/ldmud/ldmud/blob/64b3588a13fb0c761da62d6deb56dfa380b03c6f/COPYRIGHT#L16-L23) file says:

```
Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions
are met:
1. Redistributions of source code must retain the above copyright
   notice, this list of conditions and the following disclaimer.
2. Redistributions in binary form must reproduce the above copyright
   notice, this list of conditions and the following disclaimer in the
   documentation and/or other materials provided with the distribution.
```

I am not sure if 2 is met automagically by the current build, but it seems doable in some form or another. Does ensuring the COPYRIGHT file is copied into the output suffice?

cpu replied:
Using `unfreeRedistributable` seems reasonable to me as well. :+1: 

I've switched to that license in 65f142b. Separately I also made sure the COPYRIGHT file makes it into the output in 4aad58f.

dali99 replied:
I'm personally not entirely convinced this is unfreeRedistributable.

It says:

```
Permission is granted to use, extend and modify
the source code and any executables created from the source code
provided subject to the restriction that the source code and any
executables created therefrom may not be used in any way whatsoever
for monetary gain.
```

It's not clear to me that this is compatible with the redistribution clause. (For example what if someone earned money from running the cache it was served from? How direct does this have to be?) I'm not a lawyer, so idk what exactly is going on here.

but I'm personally always going to be uncomfortable with licenses which contradict themselves like this.

cpu replied:
Unfortunately the heredity of this software is quite old (the earliest form was circa ~89) and predates much of the consensus around free software licensing that has evolved since.

>  (For example what if someone earned money from running the cache it was served from? How direct does this have to be?) 

I think it will be hard to address these concerns concretely but my understanding of the context in which the non-commercial clause was added leads me to believe that it's principally concerned with games being run on LDMud with a for-profit model.

> but I'm personally always going to be uncomfortable with licenses which contradict themselves like this.

Judging by the metric of emoji reactions (the most scientific of all metrics) I think @viraptor, @abathur and myself believe we've met the bar for unfreeRedistributable. I'm going to resolve this comment but would be open to more discussion if others disagree strongly.

cpu commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,79 @@
+    sha256 = "sha256-2TaFt+T9B5Df6KWRQcbhY1E1D6NISb0oqLgyX47f5lI=";
+  };
+
+  sourceRoot = "source/src";
```
Implemented in [30182ef](https://github.com/NixOS/nixpkgs/pull/190682/commits/30182ef93e7bdc7be7625b9a0e0988b469569675).

ckiee commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,79 @@
+  installTargets = [ "install-driver" "install-utils" "install-headers" ];
+
+  postInstall = ''
+    mkdir -p "$out/share/"; cp -v ../COPYRIGHT $out/share/;
```
```suggestion
    mkdir -p "$out/share/"
    cp -v ../COPYRIGHT $out/share/
```

cpu replied:
Fixed in e651f31

ckiee commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,79 @@
+{ lib, fetchFromGitHub, stdenv, autoreconfHook, pkg-config, bison, libiconv
+, pcre, libgcrypt, ipv6Support ? false, mccpSupport ? false, zlib
+, mysqlSupport ? false, libmysqlclient, postgresSupport ? false, postgresql
+, sqliteSupport ? false, sqlite, json_c, libxml2, tlsSupport ? false, openssl
+, pythonSupport ? false, python310 }:
```
Separate the `*Support` args? See [an example](https://github.com/ckiee/nixpkgs/blob/8709490e60f42c4fcd402eb96cdfaa93f86669ae/pkgs/development/libraries/ffmpeg/generic.nix#L1-L26).

cpu replied:
Looks nicer, thanks! Fixed in e651f31

ckiee commented on pkgs/games/ldmud/default.nix:
```diff
@@ -0,0 +1,79 @@
+  '';
+
+  meta = with lib; {
+    description =
+      "LDMud is a gamedriver for LPMuds. (LPC compiler, interpreter and runtime environment.)";
```
See `CONTRIBUTING.md`.
```suggestion
    description = "A gamedriver for LPMuds including a LPC compiler, interpreter and runtime";
```

ckiee reviewed (changes requested):
Mostly styling. Seems a Sandro review comment got missed so I mentioned it; `nixpkgs-fmt`, not `nixfmt`.

cpu commented:
@ckiee Thanks for the review :-) I believe I've applied all of your suggested changes. 

> nixpkgs-fmt, not nixfmt.

When I run `nixpkgs-fmt` on this file it also splays out each of the non-Support args on their own line. Is that generally preferred? I wasn't sure so haven't added that commit yet but can do so.

ckiee commented:
If you want, yeah. Probably better to have it in a state where you can reformat the entire file and produce no diff for future changes. `nixpkgs-fmt` is kind of weird in that there's no one canonical representation and it'll no-op a variety of indents/etc.

L9 would look nicer if the `}:` was on its own line, and the `review:` commits should be squashed into `ldmud`.

cpu commented:
> If you want, yeah. Probably better to have it in a state where you can reformat the entire file and produce no diff for future changes.

Makes sense :+1: Updated with the `nixpkgs-fmt` form.

> L9 would look nicer if the }: was on its own line

`nixpkgs-fmt` agreed and made the change for me :partying_face: 

>  the review: commits should be squashed into ldmud.

:heavy_check_mark:  Rebased to just the one commit - [065000](https://github.com/NixOS/nixpkgs/pull/190682/commits/0650000314d6a3bff5d63a74f14a2094946126f6) (_strange commit hash!_ :four_leaf_clover: )

ckiee reviewed (approved):
Just noticed there's tests but it seems they're a [somewhat big pile of shell](https://github.com/ldmud/ldmud/blob/cf009941c0aefa593dc353fe62d938a3b0b4bcb5/test/run.sh). Up to you if you wanna add them since they don't seem to set up any environment and are probably only useful for the built-in stuff that should already have been ran for a release.

(Also, the commit hash is cool and it'd be cool to merge now)

cpu commented:
Thanks for the stamp Cookie.

> Just noticed there's tests but it seems they're a [somewhat big pile of shell](https://github.com/ldmud/ldmud/blob/cf009941c0aefa593dc353fe62d938a3b0b4bcb5/test/run.sh). Up to you if you wanna add them since they don't seem to set up any environment and are probably only useful for the built-in stuff that should already have been ran for a release.

I have a little bit of familiarity with this testsuite, I think it might be a little bit tricky to run here because it assumes some build options. I'll give it some thought but would prefer to try to land a follow-up PR if I get it going vs adding more to this branch since its already had a good amount of review from folks in the community.

> (Also, the commit hash is cool and it'd be cool to merge now)

Agreed! :-)
